docs/nomt_specification.md

```diff
@@ -0,0 +1,179 @@
+
+# Nearly Optimal State Merkalization - Specification
+
+## Introduction 
+
+The following document will attempt to describe each part of the specification as accurately as possible without attempting to convince the reader that this is the best structure.
```

Comment:
The part about convincing the reader seems superfluous (chatgpt?)

Reply:
ahahah, no, it's just me. I used to specify this in the past when writing design docs, but it shifted from a design doc to a specification. I agree that this part could be removed now!